File: cpp_solutions/Integer_Break.cpp
#include <iostream>
#include<string.h>
#include<math.h>
class Solution {
public:
    int integerBreak(int n) {
        if (n==2 || n==3) {return n-1;}
        if (n%3 == 2){return (pow(3,n/3)*2);}
        else if (n%3==1){return(pow(3,n/3-1)*4);}
        return pow(3,n/3);
        
    }
};

File: python_solutions/Integer_Break.py
class Solution:
    def integerBreak(self, n: int) -> int:
        if n <= 3:
            return (n - 1)
        elif n % 3 == 2:
            return (3 ** (n // 3) * 2)
        elif n % 3 == 1:
            return (3 ** (n // 3 - 1) * 4)
        else:
            return (3 ** (n // 3))

